File: app/src/main/java/mk/webfactory/template/feature/home/ui/HomePresenter.java
package mk.webfactory.template.feature.home.ui;

import javax.inject.Inject;
import mk.webfactory.template.feature.home.HomeRepository;

final class HomePresenter extends HomeContract.Presenter {

  private final HomeRepository homeRepository;

  @Inject public HomePresenter(HomeRepository homeRepository) {
    this.homeRepository = homeRepository;
  }

  @Override protected void onTakeView(HomeContract.View v) {
    if (hasView()) {
      getView().showLoadingIndicator(true);
    }
  }

  @Override protected void onDropView() {
  }
}

File: app/src/main/java/mk/webfactory/template/feature/home/ui/HomePresenter.kt
package mk.webfactory.template.feature.home.ui

import mk.webfactory.template.feature.home.HomeRepository
import javax.inject.Inject

class HomePresenter @Inject constructor(private val homeRepository: HomeRepository) :
    HomeContract.Presenter() {
    override fun onTakeView(v: HomeContract.View?) {
        if (hasView()) {
            view!!.showLoadingIndicator(true)
        }
    }

    override fun onDropView() {}

}